expr/parse.go

```diff
@@ -476,3 +476,38 @@ func aggKey(serie models.Series, nodes []expr) string {
 	}
 	return strings.Join(name, ".")
 }
+
+// filterNodesByPositions removes the node at the given position(s)
+func filterNodesByPositions(serie models.Series, nodes []expr) (string, error) {
+	metric := extractMetric(serie.Target)
+	if len(metric) == 0 {
+		return "", fmt.Errorf("invalid metric name")
+	}
+	// Trim off tags (if they are there) and split on '.'
+	parts := strings.Split(strings.SplitN(metric, ";", 2)[0], ".")
+	name := parts[:0]
```

Comment:
Why do you have all this? Graphite doesn't do this.
Graphite does only `key = '.'.join(map(lambda x: x[1], filter(lambda i: i[0] not in positions, enumerate(series.name.split('.')))))`
We need to be compatible with graphite.